main_loop_tf/main.py

```diff
@@ -719,50 +710,26 @@ def build_graph(placeholders, input_shape, optimizer, weight_decay, loss_fn,
             # During the training we want to save informations about the
             # gradients, the trainable variables and the activations.
 
-            if is_training:
-                # Merge only the training summaries of the gpus used.
-                # This is done searching by name since the variables are
-                # always preceded by 'Tower{0,1,2,..}' or 'GPU{0,1,2,..}'
-                training_summaries = []
-                for i in range(n_gpus):
-                    for el in list_summaries[idx]:
-                        if ('GPU%d' % i in el.name or
-                                'Tower%d' % i in el.name):
-                            training_summaries += [el]
-                        # if (i == 0 and
-                        #         'Mean_tower_loss_' + which_set in el.name):
-                        #     training_summaries += [el]
-                train_summary_op[idx] = tf.summary.merge(training_summaries)
-                import ipdb; ipdb.set_trace()
-            else:
-                # Same reasoning done for the training summaries (see
-                # above)
-                val_summaries = []
-                for i in range(n_gpus):
-                    for el in list_summaries[idx]:
-                        if ('GPU%d' % i in el.name or
-                            'Tower%d' % i in el.name):
-                            val_summaries += [el]
-                        # if (i == 0 and
-                        #         'Mean_tower_loss_' + which_set in el.name):
-                        #     val_summaries += [el]
-                val_summary_op[idx] = tf.summary.merge(val_summaries)
+            # Merge only the summaries of the gpus used.
+            # This is done searching by name since the variables are
+            # always preceded by 'Tower{0,1,2,..}' or 'gpu{0,1,2,..}'
+            this_summaries = []
+            for i in range(this_splits):
+                for el in summaries:
+                    if ('gpu%d' % i in el.name):
+                        this_summaries += [el]
+                    # TODO ??
```

Comment:
Ok, now I understand why you did it this way. Let's talk about it in person. It's really ugly and prone to errors to save the summaries in the main loop rather than in the graph, there has to be another way.